### ISSUE TITLE
Allow to verify method annotations

### ISSUE THREAD

**Issue by anthonyrichir** [TNG/ArchUnit#150]
I think it would be interesting to add features like being able to verify that some annotations are used (or not) on the methods of a class, for instance like this.
```java
@Test
public void check_custom_annotation_usage() {
    noClasses().that().resideOutsideOfPackage("..service..")
        .should().not().haveMethods().annotatedWith(MyCustomAnnotation.class)
        .as("Classes that reside outside package '..service..' should not use MyCustomAnnotation.class")
        .check(classes);
}
```

**Comment by codecholeric** [comment]
I'm working on pretty much that within #38. Once completed, you'll be able to write

```
methods().declaredInClassesThat().resideOutsideOfPackage("..service..")
    .should().notBeAnnotatedWith(MyCustomAnnotation.class)
```

Would that work for you?

**Comment by anthonyrichir** [comment]
Yes, it looks pretty much like what I'm looking for. Can't wait to try it :+1:

**Comment by codecholeric** [comment]
Fixed with #38
